Compiler warning-as-error for implicit narrowing of wchar_t to char in VS2019

When building the NativeExamples project from a fresh clone of the repository, I encounter this error in three places, (truncated for clarity):

```
2>C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.25.28610\include\xstring(2389,23): error C2220: the following warning is treated as an error
2>C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.25.28610\include\xstring(2400): message : see reference to function template instantiation 'void std::basic_string<char,std::char_traits<char>,std::allocator<char>>::_Construct<const wchar_t*>(_Iter,const _Iter,std::input_iterator_tag)' being compiled
2>        with
2>        [
2>            _Iter=const wchar_t *
2>        ]

...

2>C:\git\krabsetw\krabs\krabs\guid.hpp(62): message : see reference to function template instantiation 'std::basic_string<char,std::char_traits<char>,std::allocator<char>>::basic_string<std::_String_const_iterator<std::_String_val<std::_Simple_types<_Elem>>>,0>(_Iter,_Iter,const _Alloc &)' being compiled
2>        with
2>        [
2>            _Elem=wchar_t,
2>            _Iter=std::_String_const_iterator<std::_String_val<std::_Simple_types<wchar_t>>>,
2>            _Alloc=std::allocator<char>
2>        ]

...
```

The error happens when narrowing a wstring into a string in guid.hpp line 62, tdh_helpers.cpp line 104, and record_builder.hpp line 237. For example, from guid.hpp:

```C++
inline guid::guid(const std::wstring &guid)
    {
            ...
            std::string guidStr(guid.begin(), guid.end());
            ...
        }
    }
```

To fix the issue for my own purposes, I explicitly wrapped the line:

```C++
#pragma warning( push )
#pragma warning( disable : 4244 )
            std::string guidStr(guid.begin(), guid.end());
#pragma warning( pop )
```

Based on what I can see, in all three usages based on how the code is eventually invoked, it's probably safe to assume that the wchar's can be narrowed to just chars, but it'd be nice if this didn't create compiler errors out of the box. :)

My repro steps:

1. Clone the repo, update nuget packages, etc.
2. Open with VS2019
3. Build solution (Had to rebuild when I had cached build outputs).

--- a/krabs/krabs/guid.hpp
+++ b/krabs/krabs/guid.hpp
@@ -59,7 +59,10 @@ namespace krabs {
     {
         HRESULT hr = CLSIDFromString(guid.c_str(), &guid_);
         if (FAILED(hr)) {
+#pragma warning(push)
+#pragma warning(disable: 4244) // narrowing guid wchar_t to char for this error message
             std::string guidStr(guid.begin(), guid.end());
+#pragma warning(pop)
             std::stringstream stream;
             stream << "Error in constructing guid from string (";
             stream << guidStr;

--- a/krabs/krabs/tdh_helpers.hpp
+++ b/krabs/krabs/tdh_helpers.hpp
@@ -101,7 +101,10 @@ namespace krabs {
 
             if (requested == actual) return;
 
+#pragma warning(push)
+#pragma warning(disable: 4244) // narrowing property name wchar_t to char for this error message
             std::string ansiName(name.begin(), name.end());
+#pragma warning(pop)
 
             throw type_mismatch_assert(
                 ansiName.c_str(),

--- a/krabs/krabs/testing/record_builder.hpp
+++ b/krabs/krabs/testing/record_builder.hpp
@@ -233,8 +233,12 @@ namespace krabs { namespace testing {
         if (!results.second.empty()) {
             std::string msg = "Not all the properties of the event were filled:";
 
-            for (auto& s : results.second)
+            for (auto& s : results.second) {
+#pragma warning(push)
+#pragma warning(disable: 4244) // narrowing property name wchar_t to char for this error message
                 msg += " " + std::string(s.begin(), s.end());
+#pragma warning(pop)
+            }
 
             throw std::invalid_argument(msg);
         }

